src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -339,6 +341,31 @@ private static void AddResponseContentTypes(IList<ApiResponseFormat> apiResponse
             }
         }
 
+        private static void AddActionDescriptorEndpointMetadata(
+            ActionDescriptor actionDescriptor,
+            EndpointMetadataCollection endpointMetadata)
+        {
+            if (endpointMetadata.Count > 0)
+            {
+                // ActionDescriptor.EndpointMetadata is an empty array by
+                // default so need to add the metadata into a new list.
+                var metadata = new List<object>(endpointMetadata.Count);
```

Comment:
Why not?

```suggestion
                var metadata = new List<object>(endpointMetadata);
```

Is filtering out `null` important? What's adding `null`?

Reply:
Just a habit, have simplified.